main/options.c

```diff
@@ -871,15 +872,18 @@ static void cArgRead (cookedArgs *const current)
 		item = argItem (current->args);
 		current->shortOptions = NULL;
 		Assert (item != NULL);
-		if (strncmp (item, "--", (size_t) 2) == 0)
+
+		if ((!NonOptionEncountered)
+			&& (strncmp (item, "--", (size_t) 2) == 0))
```

Comment:
Doesn't using the already existing `NonOptionEncountered` (which IIUC gets set to true when matching a non-option) break passing options after a file name altogether?  This would be unfortunate because for now it works fine, and there's even special handling for *some* options that should be passed before filenames.

Reply:
@b4n, you are correct.
Furthermore, it seems that I broke the compatibility of the command line interface in the past.

Exuberant-ctags:
```
[jet@localhost]~/var/ctags% ctags  -o - foo.c --list-kinds=Sh          
f  functions
```

Univeresal-ctags:
```
[jet@localhost]~/var/ctags% ./ctags  -o - foo.c --list-kinds=Sh
ctags: Warning: cannot open input file "--list-kinds=Sh" : No such file or directory
a	foo.c	/^void a () {}$/;"	f	typeref:typename:void
b	foo.c	/^void b () {}$/;"	f	typeref:typename:void
```

U-ctags ignores the option (--list-kinds) after input file name.

Another example:
```
[jet@localhost]~/var/ctags% ctags  -o - foo.c --sort=no      
b	foo.c	/^void b () {}$/;"	f
a	foo.c	/^void a () {}$/;"	f
[jet@localhost]~/var/ctags% ./ctags  -o - foo.c --sort=no      
ctags: Warning: cannot open input file "--sort=no" : No such file or directory
a	foo.c	/^void a () {}$/;"	f	typeref:typename:void
b	foo.c	/^void b () {}$/;"	f	typeref:typename:void
```

This incompatibility is nothing to do with the newly introduced `--` option.
I will do bisect and spot the change that introduced the incompatibility first.
After that, I will revise this pull request.

Reply:
I will work again on this item after reivising #1878.